tools/text_processing/text_processing/awk.xml

```diff
@@ -1,48 +1,126 @@
+        #end for
         awk
             --sandbox
             -v FS='	'
             -v OFS='	'
             --re-interval
-            -f "$awk_script"
-            "$infile"
-        > "$outfile"
+            -f '$awk_script'
+            #for group in $infiles
+                tool_input=$tool_input
+                #set $tool_input+=1
+                #if $group.input_type_selection.multiple
+                    #set $inputs = $group.input_type_selection.infile
+                    ## Allow capturing the collection id in the program
+                    ## tool_input_id='$inputs.element_identifier'  FIXME collections dont expose their label to tools.
+                #else
+                    #set $inputs = [$group.input_type_selection.infile]
+                #end if
+                #for file in $inputs
+                    tool_input_id='$file.element_identifier'
+                    '$file'
+                #end for
+            #end for
+        > '$outfile'
 ]]>
     </command>
     <configfiles>
         <configfile name="awk_script">$code</configfile>
     </configfiles>
     <inputs>
-        <param name="infile" format="txt" type="data" label="File to process" />
+        <repeat name="infiles" title="Inputs">
+            <conditional name="input_type_selection">
+                <param name="multiple" type="boolean" checked="false" label="Accept multiple" />
+                <when value="false">
+                    <param name="infile" format="txt" type="data" label="Single file to process" help="A separate execution will occur for each provided file"/>
+                </when>
+                <when value="true">
+                    <param name="infile" format="txt" type="data" multiple="true" label="File or collection to process" />
+                </when>
+            </conditional>
+        </repeat>
         <param name="code" type="text" area="true" size="5x35" label="AWK Program" help="">
-            <sanitizer>
-                <valid initial="string.printable">
-                    <remove value="&apos;"/>
-                </valid>
-            </sanitizer>
+            <sanitizer sanitize="false" />
         </param>
+        <repeat name="envs" title="Environment Variables">
+            <param name="name" type="text" label="Name">
+                <sanitizer>
+                    <valid initial="string.printable">
+                        <remove value="&apos;" />
+                    </valid>
+                </sanitizer>
+            </param>
+            <param name="val" type="text" label="Value">
+                <sanitizer>
+                    <valid initial="string.printable">
+                        <remove value="&apos;" />
+                    </valid>
+                </sanitizer>
+            </param>
+        </repeat>
     </inputs>
     <outputs>
-        <data name="outfile" format_source="infile" metadata_source="infile"/>
+        <data name="outfile" format="txt" />
```

Comment:
I think this is not a good idea, e.g. if you input a tabular.

Reply:
What do you mean with dynamic inputs?

Reply:
You can have an output for each case of the input conditional and filter them based on the chosen input option.

Reply:
> it boiled down to Galaxy not supporting referring to inputs within a repeat block

Just one more reason to get rid of the repeat block

Reply:
Repeat blocks are the only way to have a variable number of inputs.